### PR TITLE
CP-51379: Implement smb resource for file based storage

### PR DESCRIPTION
**Pull request by xueqingz**
test result :
```
ubuntu@ubuntu-server:~/code/terraform-provider-xenserver$ make testacc 
if [ -z "/home/ubuntu/go/bin" ]; then echo "GOBIN is not set" && exit 1; fi
go mod tidy
go install .
ls -l /home/ubuntu/go/bin/terraform-provider-xenserver
-rwxrwxr-x 1 ubuntu ubuntu 23307403 Sep 13 02:58 /home/ubuntu/go/bin/terraform-provider-xenserver
source .env && TF_ACC=1 go test ./xenserver/ -v   -timeout 120m
=== RUN   TestAccNetworkDataSource
--- PASS: TestAccNetworkDataSource (0.70s)
=== RUN   TestAccVlanResource
--- PASS: TestAccVlanResource (3.71s)
=== RUN   TestAccNICDataSource
--- PASS: TestAccNICDataSource (0.64s)
=== RUN   TestAccPifDataSource
--- PASS: TestAccPifDataSource (1.16s)
=== RUN   TestAccSnapshotResource
--- PASS: TestAccSnapshotResource (6.23s)
=== RUN   TestAccSRDataSource
--- PASS: TestAccSRDataSource (0.65s)
=== RUN   TestAccNFSResource
--- PASS: TestAccNFSResource (4.75s)
=== RUN   TestAccSRResourceLocal
--- PASS: TestAccSRResourceLocal (4.56s)
=== RUN   TestAccSRResourceShared
--- PASS: TestAccSRResourceShared (4.07s)
=== RUN   TestAccSMBResource
--- PASS: TestAccSMBResource (7.11s)
=== RUN   TestAccVDIResource
--- PASS: TestAccVDIResource (5.53s)
=== RUN   TestAccVMDataSource
--- PASS: TestAccVMDataSource (2.43s)
=== RUN   TestAccVMResource
--- PASS: TestAccVMResource (4.61s)
=== RUN   TestAccLinuxVMResource
--- PASS: TestAccLinuxVMResource (2.35s)
PASS
ok      terraform-provider-xenserver/xenserver  48.515s
```

### CODE REVIEW

File: docs/resources/sr_smb.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `name_label` (String) The name of the SMB storage repository.
+- `storage_location` (String) The server and server path of the SMB storage repository.<br />Follow the format `"\\server\path"`.
```
Comment (acefei) [inline docs/resources/sr_smb.md:31]: Do we need to consider escaping with the '\'?

**Review by JinyuZhou-patty — APPROVED** [review]

File: docs/resources/sr_smb.md (hunk 1 above)
Comment (xueqingz) [reply docs/resources/sr_smb.md:31]: Right, let me update there.

**Review by JinyuZhou-patty — APPROVED** [review]

File: docs/resources/sr_smb.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+Provides an SMB storage repository resource.
+
+## Example Usage
+
+```terraform
+resource "xenserver_sr_smb" "smb_test" {
+  name_label       = "SMB storage"
+  name_description = "A test SMB storage repository"
+  storage_location = "\\\\server\\path"
```
Comment (acefei) [inline docs/resources/sr_smb.md:19]: We can add an example for escaped path as follows:
```
 storage_location = <<-EOF
    \\server\path
  EOF
  ```

Comment (xueqingz) [reply docs/resources/sr_smb.md:19]: It store like "\\\\server\\path\n" in state file, after discussion with @JinyuZhou-patty , we don't want it.

Comment (xueqingz) [reply docs/resources/sr_smb.md:19]: Or we can just add an extra example. Let me update the examples.

**Review by acefei — APPROVED** [review]

**Pull request merged**